### DISCUSSION TITLE
S3 access problem

### DISCUSSION THREAD

**Discussion by skrey** [seaweedfs/seaweedfs#2980] [Q&A] [answered]
Hi.
I would like to test SeasweedFS and evaluate, if it is a possible candidate for us, but I am having problems accessing the system via S3.

I have started the server with:  ./weed server -s3 -dir=./ -master.port=9333

Then created credentials in weed shell with: s3.configure -access_key=xxxxxx -secret_key=yyyyyy -user=skrey -actions=Read,Write,List,Tagging,Admin -apply

And then I have tried to run warp or list the contents of the warp-benchmark-bucket, but get this error:

[skrey@skrey-p5530 seaweed]$ ./warp mixed --host localhost:8888 --access-key=xxxxxx --secret-key yyyyyy
Creating Bucket "warp-benchmark-bucket"...                                                                    
warp: <ERROR> Error preparing server. XML syntax error on line 79: element <link> closed by </head>.
[skrey@skrey-p5530 seaweed]$ rclone ls S3local:warp-benchmark-bucket
2022/04/27 11:30:16 Failed to ls: SerializationError: failed to decode REST XML response
	status code: 200, request id: 
caused by: XML syntax error on line 79: element <link> closed by </head>

The web interface of the filer shows the created bucket. What is going wrong here?

Thank you in advance.

Best regards,
Sebastian

**Comment by kmlebedev** [comment, marked as answer]
It is obvious that the matter is in the xml response. 
Do you have the ability to turn on the debug log in such a way that you can see the entire xml response?

**Reply by skrey** [reply]
Sorry, did not write a reply, just a new answer.
Any recommendation for a S3 client that can dump the raw response?

This is the rclone debug output.
```
[skrey@skrey-p5530 seaweed]$ rclone -vvv ls S3local:warp-benchmark-bucket
2022/04/27 12:22:30 DEBUG : rclone: Version "v1.58.0" starting with parameters ["rclone" "-vvv" "ls" "S3local:warp-benchmark-bucket"]
2022/04/27 12:22:30 DEBUG : Creating backend with remote "S3local:warp-benchmark-bucket"
2022/04/27 12:22:30 DEBUG : Using config file from "/home/skrey/.config/rclone/rclone.conf"
2022/04/27 12:23:06 DEBUG : S3 bucket warp-benchmark-bucket: Retrying listing because of characters which can't be XML encoded
2022/04/27 12:23:06 DEBUG : pacer: low level retry 1/2 (error SerializationError: failed to decode REST XML response
	status code: 200, request id: 
caused by: XML syntax error on line 79: element <link> closed by </head>)
2022/04/27 12:23:06 DEBUG : pacer: Rate limited, increasing sleep to 10ms
2022/04/27 12:23:46 DEBUG : pacer: Reducing sleep to 0s
2022/04/27 12:23:46 DEBUG : 2 go routines active
2022/04/27 12:23:46 Failed to ls: SerializationError: failed to decode REST XML response
	status code: 200, request id: 
caused by: XML syntax error on line 79: element <link> closed by </head>
```

**Reply by skrey** [reply]
I have found my mistake. I have connected to the filer port 8888 instead of the s3 port 8333.

**Reply by zyjnd** [reply]
> I have found my mistake. I have connected to the filer port 8888 instead of the s3 port 8333.

me too，thanks to your share
